### DISCUSSION TITLE
otInstanceReset(Instance)  Function problem

### DISCUSSION THREAD

**Discussion by tuopu** [openthread/openthread#8371] [Q&A] [answered]
An exception occurs. When calling otInstanceReset (Instance), the program cannot be reset. During the Linux simulation, the program will exit and cannot continue to run.

**Comment by jwhui** [comment, marked as answer]
The CLI example app implements the [`reset`](https://github.com/openthread/openthread/blob/main/src/cli/README.md#reset) command using `otInstanceReset()`:

https://github.com/openthread/openthread/blob/ba826ffb2ac7e2d366ebd553b92a912434ff603d/src/cli/cli.cpp#L248

I just tested it with a simulation build and it seems to work as expected.

**Reply by tuopu** [reply]
In the same case, I actually use this function in another process bar, and it is invalid, that is, it must be in the same thread.
